File: 프로그래머스/unrated/155652. 둘만의 암호/둘만의 암호.cpp
#include <string>
#include <vector>

using namespace std;

string solution(string s, string skip, int index) {
    string answer = "";
    for (int i = 0; i < s.size(); i++) {
        int temp = s[i];
        int k = index;
        while (k--) {
            if (temp + 1 <= 'z')
                temp++;
            else
                temp -= 25;
            if (skip.find(temp) != string::npos) k++;
        }
        answer += temp;
    }
    return answer;
}

File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.java
import java.util.*;

class Solution {
    public String solution(String s, String skip, int index) {
        Set<Character> set = new HashSet<>();
        
        for (char chars : skip.toCharArray()) {
            set.add(chars);
        }
        
        char[] chars = s.toCharArray();
        for (int i = 0; i < s.length(); i++) {
            char c = chars[i];
            for (int j = 0; j < index; j++) {
                do {
                    c++;    
                    if (c > 'z') {
                      c = 'a';
                    }
                } while (set.contains(c));
   
            }
            chars[i] = c;
        }
        
        return new String(chars);
    }
}